extensions/reg-core-3.3.md

```diff
@@ -0,0 +1,224 @@
+authentication layer.  It is similar to current methods of signalling that intent.
+
+A `REG CREATE` command consists of the following format:
+
+    REG CREATE <accountname> [callback_namespace:]<callback> [cred_type] :<credential>
+
+The `credential` field specifies a credential of `cred_type`.  If there is no specified credential, then
+the `credential` field MUST be `*`.  A passphrase type `credential` MAY have spaces in it. New credential
+types may or may not allow spaces inside `credential`.
+
+The `callback` field designates an opaque value which indicates where a verification code, if any,
+will be sent.  The callback field is implementation-specific, with namespaces being listed using the
+`REGCALLBACKS` ISUPPORT token.  An invalid callback parameter should result in the `ERR_REG_INVALID_CALLBACK`
+error.  In the event that a callback is not provided, the client SHOULD send `*` to indicate that
+they are not providing a callback resource.  If a callback namespace is not explicitly provided, the IRC
+server MAY choose to use `mailto` as a default.
+
+The `cred_type` field designates a value which indicates the type of supplied credential.  The accepted
+values of the `cred_type` field is implementation-specific, with credential types being listed using the
+`REGCREDTYPES` ISUPPORT token.  An invalid credential type parameter should result in the `ERR_REG_INVALID_CRED_TYPE`
+error.  The client SHOULD supply a `cred_type` value, however if one is not provided the IRC server SHOULD
+use `passphrase` as a default credential type.
+
+The IRC server MAY forward the `REG CREATE` command to a central authority, or process it locally. Clients
+SHOULD NOT be sent unnecessary legacy (e.g. from NickServ) private messages or notices in response to `REG`
+commands, where native numerics such as the ones defined in this document can replace them.
+
+Upon success, the IRC server MUST send the `RPL_REGISTRATION_SUCCESS` numeric, which looks like:
+
+| No. | Label                      | Format                                                         |
+| --- | -------------------------- | -------------------------------------------------------------- |
+| 920 | `RPL_REGISTRATION_SUCCESS` | `:<server> 920 <user_nickname> <accountname> :Account created` |
+
+If the server requires a verification token, it MUST also reply with the `RPL_REG_VERIFICATION_REQUIRED`
+numeric, which looks like:
+
+| No. | Label                           | Format                                                                                                       |
+| --- | ------------------------------- | ------------------------------------------------------------------------------------------------------------ |
+| 927 | `RPL_REG_VERIFICATION_REQUIRED` | `:<server> 927 <user_nickname> <accountname> <callback_namespace:><callback> :A verification token was sent` |
+
+Upon error, the IRC server MUST send an error code that is relevant.  We suggest these numerics:
```

Comment:
Could you add the requirement for the error code to be in a given list?
Otherwise, servers may send something a bot does not expect to be a reply to this, and the bot will not know what to do with it.

Reply:
Resolved now